### ISSUE TITLE
Fix series/show

### ISSUE THREAD

**Issue by Noxville** [datdota/datdota#23]
The show action on series controller is broken or needs to be migrated to the new API system.

**Comment by hnegus1** [comment]
Hello. Has there been any progress on this? Thanks.

**Comment by Noxville** [comment]
Basically I'm not happy with the data provided by Valve's API on this:
- it's human generated and there's loads of errors
- bo2 and bo3 are using the same enum/identifier
- only started at some point in time 

I've come up with some code which I think will do a better job at series identification automatically. Still testing it out, but we had a discussion on discord about it and seems solid.

**Comment by hnegus1** [comment]
I pretty much have the same problems regarding Valve's API. It's a pain having to manually track series!

For DPC events + TI, it seems there _is_ proper distinction between BO2s and other series types through the [GetLeagueData](https://steamapi.xpaw.me/#IDOTA2League) call under the node groups ([example](https://www.dota2.com/webapi/IDOTA2League/GetLeagueData/v0001/?league_id=10749)), but it seems to be just for DPC events that they do it and I'm not even sure if Valve want us to use the call since it's not documented. 

Thanks for the update!

**Comment by Noxville** [comment]
Yeah I remember looking at this when it was added. The reason I've not yet run my 'Series Guesser' code is that I've got a fair few broken team_appearances (a match, team join-table) which I'd like to fix up before running it, otherwise detecting series in a league between two teams will just break if any individual games are broken.

```
 tier |  sum  |          error_rate           
------+-------+------------------------
    1 | 50235 | 0.05482233502538071066
    2 | 33878 | 0.09649329948639234902
    3 | 46188 | 0.15504027020005196155
    4 | 30524 | 0.53086096186607259861
```
There are two key reasons why teams are set incorrectly:

1. Valve deleted some teams, and removed them from the API for some unknown reason. This means if you look for some matches the radiant_team_id or dire_team_id for that team just isn't set. 
2. A team/admin forgot to get the team set before the match started.

#1 is correctable if I have the replay and a parser, since I can extract the team data directly from it (i.e. all of source 2). #2 isn't automatically correctable, but since these are more likely temporary issues (admin/team error), you can look at surrounding matches for that exact squad to rectify many. I want tier 1 and 2 (premium + pro) to be under ~2.5% error rate before I make the series guesser live.

```
 tier | source |  sum  |          avg           
------+--------+-------+------------------------
    1 | s1     | 19045 | 0.10207403517983722762
    1 | s2     | 31190 | 0.02596986213529977557
    2 | s1     | 19518 | 0.11988933292345527206
    2 | s2     | 14360 | 0.06469359331476323120
    3 | s1     | 20852 | 0.19513715710723192020
    3 | s2     | 25336 | 0.12203978528575939375
    4 | s1     |  2444 | 0.38256955810147299509
    4 | s2     | 28080 | 0.54376780626780626781

```
The guesser has a category BO2_OR_3 and then it's easy to find and manually verify blocks of those based on the tournament and surrounding matches.

**Comment by Noxville** [comment]
Made some improvements to this error rate. Still have 1 more idea to run before being happy with it, but an improvement:

 ```
tier | source | team_appearances | broken_team |          avg           
------+--------+------------------+-------------+------------------------
    1 | s1     |            19045 |        1767 | 0.09278025728537673930
    1 | s2     |            31134 |         303 | 0.00973212565041433802
    2 | s1     |            19516 |        2097 | 0.10745029719204755073
    2 | s2     |            14334 |         316 | 0.02204548625645318822
    3 | s1     |            20852 |        4069 | 0.19513715710723192020
    3 | s2     |            25338 |        3092 | 0.12203015234035835504
    4 | s1     |             2444 |         935 | 0.38256955810147299509
    4 | s2     |            28080 |       15269 | 0.54376780626780626781
```

**Comment by Noxville** [comment]
Okay, so I now have a 'DerivedSeries' entity associated with each match based on derived data (same teams playing within a threshold in the same league without playing other teams inbetween). Here's the derived series length -> frequency map for pro matches.

 ```
len | sum  |    array     
-----+------+--------------
   1 | 9988 | {}
   2 | 9911 | {}
   3 | 3543 | {}
   4 |  223 | {}
   5 |  148 | {}
   6 |    2 | {2409,14312}
   7 |    1 | {4168}
(7 rows)

```

This is also referenced in the API (as derived_series).

There are a few minor bugs and things to do:

1. Make the job run every ~12 hours or so.
2. Make another job which sets/guesses bo2 vs bo3.
3. Source1 remake games annoyingly bug this out. There's no real fix there.
